### PR TITLE
Add qByteArray to script value conversion

### PR DESCRIPTION
**Pull request by ksuprynowicz**
This will be useful for sending binary data such as audio streams to and from the script engine.

### CODE REVIEW

**Comment by ksuprynowicz** [comment]
After reviewing and merging this we can work on audio streaming support

File: libraries/script-engine/src/ScriptManagerScriptingInterface.h [hunk 1]
```diff
@@ -581,6 +581,22 @@ class ScriptManagerScriptingInterface : public QObject {
      Q_INVOKABLE void removeServerEntityScriptMessagesRequest();
      Q_INVOKABLE void removeServerEntityScriptMessagesRequest(const QUuid& entityID);
 
+     /*@jsdoc
+     * This decodes Base64 string and returns contents as ArrayBuffer.
+     * @function Script.atob
+     * @param {String} base64 - String with Base64-encoded binary data.
+     * @returns {ArrayBuffer} Decoded binary data.
+     */
+     Q_INVOKABLE QByteArray atob(const QString &base64);
```
Comment (daleglass) [inline libraries/script-engine/src/ScriptManagerScriptingInterface.h:590]: Is there a reason to give it such a terse name?

**Review by daleglass — CHANGES_REQUESTED** [review]

File: libraries/script-engine/src/ScriptManagerScriptingInterface.h (hunk 1 above)
Comment (ksuprynowicz) [reply libraries/script-engine/src/ScriptManagerScriptingInterface.h:590]: Yes, I'd prefer more descriptive name, but this one is standard for JS:
https://developer.mozilla.org/en-US/docs/Web/API/atob

**Review by daleglass — APPROVED** [review]

**Comment by ksuprynowicz** [comment]
I tested it, I think it's safe to merge

**Pull request merged**